cc/sgf.cc

```diff
@@ -219,6 +220,44 @@ bool GetTreeImpl(const Ast::Tree& tree,
       move.color = Color::kBlack;
     } else if ((prop = node.FindProperty("W")) != nullptr) {
       move.color = Color::kWhite;
+    } else if ((prop = node.FindProperty("HA")) != nullptr) {
+      int handicap;
+      bool valid_handicap = absl::SimpleAtoi(prop->values[0], &handicap);
+      if (!valid_handicap) {
+        MG_LOG(ERROR) << "Invalid handicap property: " << prop->values[0];
+        break;
+      }
+
+      if ((handicap <= 1) || (handicap > 9)) {
+        MG_LOG(ERROR) << "Invalid handicap value:" << handicap;
+        return false;
+      }
+      stones = node.FindProperty("AB");
+      if (stones == nullptr) {
+        MG_LOG(ERROR) << "Handicap stones not specified.";
+        return false;
+      }
+
+      for (const auto& h_location : stones->values) {
+        Move m;
+        m.color = Color::kBlack;
+        m.c = Coord::FromSgf(h_location, true);
+        if (m.c == Coord::kInvalid) {
+          MG_LOG(ERROR) << "Can't parse node " << node.ToString() << ": \""
+                        << h_location << "\" isn't a valid SGF coord for a handicap stone";
+          return false;
+        }
+        std::string comment;
+        dst->push_back(absl::make_unique<Node>(m, std::move(comment)));
+        dst = &(dst->back()->children);
+        if (h_location != stones->values.back()) {
+          m.color = Color::kWhite;
+          m.c = Coord::kPass;
+          dst->push_back(absl::make_unique<Node>(m, std::move(comment)));
```

Comment:
same here (this is actually undefined behaviour because you need to reinitialized the moved-from `comment` before using it again. Clang would have probably warned about this)

Reply:
Done.